src/components/Auth/Auth.js

```diff
@@ -0,0 +1,388 @@
+  TextField,
+  Grid,
+  IconButton,
+  Button
+} from '@material-ui/core';
+import { ArrowBack } from '@material-ui/icons';
+//import { useDispatch } from 'react-redux';
+//import axios from '../Api/Api';
+
+export default function Auth() {
+  // Dispatch for react-redux store
+  // const dispatch = useDispatch();
+  // Local state control of displaying sign-in or sign-up info
+  const [mainVisible, setMainVisible] = useState(true);
+  const [mainDirection, setMainDirection] = useState('left');
+  const [signUpVisible, setSignUpVisible] = useState(false);
+  const [signUpDirection, setSignUpDirection] = useState('left');
+  const [signInVisible, setSignInVisible] = useState(false);
+  const [signInDirection, setSignInDirection] = useState('left');
+
+  // Store user's information in React state
+  const [username, setUsername] = useState("");
+  const [usernameError, setUsernameError] = useState(false);
+  const [usernameErrorMsg, setUsernameErrorMsg] = useState('');
+  const [password, setPassword] = useState("");
+  const [passwordError, setPasswordError] = useState(false);
+  const [passwordErrorMsg, setPasswordErrorMsg] = useState('');
+  const [email, setEmail] = useState("");
+  const [emailError, setEmailError] = useState(false);
+  const [emailErrorMsg, setEmailErrorMsg] = useState('');;
+
+  // Show main modal which gives user the option to signin or signup
+  const showMain = () => {
+    setMainDirection('left');
+    setMainVisible(true);
+    setSignUpVisible(false);
+    setSignUpDirection('right');
+    setSignInVisible(false);
+    setSignInDirection('right');
+  };
+
+  // Handles showing the signup window
+  const showSignUp = () => {
+    setSignUpDirection('left');
+    setMainDirection('right');
+    setSignUpVisible(true);
+    setMainVisible(false);
+    setSignInDirection('right');
+    setSignInVisible(false);
+    setUsernameErrorMsg('');
+    setUsernameError(false);
+    setPasswordErrorMsg('');
+    setPasswordError(false);
+    setEmailErrorMsg('');
+    setEmailError(false);
+  };
+
+  // Handles showing the signIn window
+  const showSignIn = () => {
+    setSignInDirection('left');
+    setMainDirection('right');
+    setSignInVisible(true);
+    setMainVisible(false);
+    setSignUpDirection('right');
+    setSignUpVisible(false);
+    setUsernameErrorMsg('');
+    setUsernameError(false);
+    setPasswordErrorMsg('');
+    setPasswordError(false);
+    setEmailErrorMsg('');
+    setEmailError(false);
+  };
+
+
+  // Handles and checks keypress and calls the callback method
+  const handleKeyPress = (e, callBack) => {
+    if (e.key === 'Enter') {
+      callBack();
+    };
+  };
+
+  // Validates user entered sign up information correctly
+  const handleOnSignUpSubmit = (username, password, email, callBack) => {
+    let error = false;
+    if (username === '') {
+      setUsernameError(true);
+      setUsernameErrorMsg('Username cannot be empty');
+      error = true;
+    } else setUsernameError(false);
```

Comment:
Nit, I'd keep the bracket style consistent in all files. This applies everywhere you use else in this file.

```
if {
  ...
} else {
  ...
}
```

Reply:
Fixed along with removing the backup files from the commit

Reply:
I think when you made edits, you added back the backup ~ files. Please remove them before submitting.

Also, not seeing the else changes. Basically this is what I was looking for:
```
   if (username === '') {
      setUsernameError(true);
      setUsernameErrorMsg('Username cannot be empty');
      error = true;
    } else {
      setUsernameError(false);
   };
```

Reply:
The changes are in new commits specifically eb392cb and 6ee58f6